### ISSUE TITLE
Nao does not work when NVDA runs from source

### ISSUE THREAD

**Issue by CyrilleB79** [sharkboyto/nao#21]
Hello

Here is a bug report. Thanks to all for the great work on this add-on!

## Issue description
When I run NVDA from source, I get the following error at initialization:

```
ERROR - globalPluginHandler.listPlugins (10:35:54.260) - MainThread (6672):
Error importing global plugin nao
Traceback (most recent call last):
  File "globalPluginHandler.py", line 23, in listPlugins
    plugin = importlib.import_module("globalPlugins.%s" % name, package="globalPlugins").GlobalPlugin
  File "C:\Users\Cyrille\AppData\Local\Programs\Python\Python37-32\lib\importlib\__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "C:\Users\Cyrille\AppData\Roaming\nvda\addons\nao\globalPlugins\nao\__init__.py", line 15, in <module>
    from .framework.generic.updates import AutoUpdates, ManualUpdatesCheck
  File "C:\Users\Cyrille\AppData\Roaming\nvda\addons\nao\globalPlugins\nao\framework\generic\updates.py", line 24, in <module>
    from .http import json_post
  File "C:\Users\Cyrille\AppData\Roaming\nvda\addons\nao\globalPlugins\nao\framework\generic\http.py", line 12, in <module>
    import updateCheck
  File "updateCheck.py", line 20, in <module>
    raise RuntimeError("No update version type, update checking not supported")
RuntimeError: No update version type, update checking not supported

```

Thus Nao feature are not available.

## Expected result

Nao plugin should be able to operate also when NVDA is run from source.

If there is ever an incompatibility with the update feature of the add-on, only the update feature should be made unavailable. A warning should appear in the log rather than an error which prevent the future execution of the global plugin.

I wonder however if the update feature is really incompatible. If I understand correctly, you call NVDA's updateCheck just for the certificate issue. But this issue should be fixed with NVDA 2021.3 onwards.

**Comment by davidedec** [comment]
Thanks for opening this.
We will check for a solution as we are keeping compatibility since NVDA 2019.3

**Comment by davidedec** [comment]
@CyrilleB79 
If you are able to make NAO from sources, please clone the develop branch and try again

**Comment by CyrilleB79** [comment]
I have just tested with Nao's develop branch (commit dc06623d9fb2a25443eb6695db433afb5809e3ed) and NVDA run from source, master branch (commit b5f82f878f344ab26004f39a291da5d65a57efb0).
I have built myself Nao from the cloned repo running "scons dev=True" and installed it.
Then I have run NVDA from source using the "-c" flag in the command line in order to point to my usual NVDA config where Nao is installed.

Here are the results:
* No more error in the log at NVDA (and add-on startup)
* The add-on's scripts (NVDA+shift+R in Windows Explorer and NVDA+shift+control+R) are working as expected.
* Trying NVDA menu -> Tools -> Nao -> Search for update, I get an error message box. Is it expected? The message box text, approximately translated back to English is:
  Nao update - Error
  Error during update search
* The add-on has not asked fo for automatic update. I do not know if it is supposed to. Since I have used the dev=True flag when building Nao, the version of the add-on is 20220105-dev.

I do not know if you can and/or want to improve the issues related to add-on update. Anyway, on my side, I consider the issue fixed by this commit; as explained before, I do not expect the update mechanism to work when running NVDA from source.

**Comment by davidedec** [comment]
Thanks for your test.
The error of the update is due to the add-on version 20220105-dev that doesn't conform to NAO standard. It fails as it can't get the correct version to compare.
We will not consider this an issue at the moment.
Thank you again

**Comment by CyrilleB79** [comment]
Hello

For test purpose, I have modified manually the version in the manifest to 2022.1 and restarted NVDA.
Then, with NVDA running from source, I have searched for update.
The update to 2022.1.1 was available and I have been able to update successfully to it.
I have not tested automatic update.

It has not updated to the last "develop" version, but I guess it is intended.

To summarize, update mechanism is also working when running from source.
